src/utilities/theme/role-variants.ts

```diff
@@ -574,8 +572,7 @@ export const roleVariants: RoleVariants = {
       name: 'decorativeTwoSurface',
       description: 'For use as a decorative surface color.',
       light: {
-        hue: 30,
-        saturation: 100,
+        hue: (hue) => (((hue - 26) % 360) + 360) % 360,
```

Comment:
Now this take in the hue from the base decorative color and rotates it by -26 degrees. The modulo operations here are just a fancy one-line way to keep the number between 0 and 360 and wrap according. For example, 370 would become 10 and -10 would become 350.

Reply:
I feel like these magic number rotations should be constants i.e. `orangeHueRotation` or something descriptive like that.

Reply:
Curious why you feel the `-26` relative adjustment is different than say a fixed adjustment of `84`? The unit here is degrees and the starting degree of the base color would be unknown so color related names wouldn’t work.

I would instead like the update the role variant names to reflect something about the color, rather than 1, 2, 3, 4, and 5. But so far haven’t come up with any ideas other than adding the degree rotation to the variant name, for example `surfaceDecorativeBase`, `surfaceDecorative88`, `surfaceDecorative124`, `surfaceDecorative308`, `surfaceDecorative334`. Those are brittle though, if we tweak the relative hue adjustment that means we should update the name.

Reply:
I realized after commenting that a color based name wouldn't work. I originally thought basing the rotation off of the original colour to the new color would be more descriptive but it really wouldn't. 

It's hard to tell what `-26` is doing here at a glance without knowing the lib so I think adding the word `rotation` or `paletteRotation` might help with that. You're right about the fragility of adding a number to the constant though. 

I'm perfectly fine with shipping as is

Reply:
I think `(((hue - 26) % 360) + 360) % 360` can be simplified to `(360 + hue - 26) %360`.

What if we did some currying clever-bollocks:


```js
// A function that returns a function that rotates the hue by a certain amount
function hueRotationFn(rotation) {
return (hue) => (360 + hue + rotation) % 360;
}


// Then in the variants:
hue: hueRotationFn(-26),
```

That way the clamping is encapsulated within the function factory rather than being duplicated on every line. You should be able to do a similar thing with the saturation clamping too

```js
function saturationOffsetFn(offset) {
  // Do we want to clamp this to an upper bound too?
  return (sat) => Math.max(sat + offset, 0),
}
```

Reply:
> I think (((hue - 26) % 360) + 360) % 360 can be simplified to (360 + hue - 26) % 360.

Nice, good catch

Reply:
> rather than being duplicated

Yeah I intentionally wanted to stay away from any dependencies for the lambdas, even if that was a function in the same file. Although that’s a gut feel and having a hard time putting that into words.

Reply:
> Do we want to clamp this to an upper bound too?

It isn’t needed because the only adjustments made are subtractions. Though if abstracting it for general use, yes it should be clamped on the upper bound.